backend/src/services/import_service.py

```
@@ -0,0 +1,326 @@
+                        data[entity_type] = json.loads(entity_data.decode('utf-8'))
+                    except KeyError:
+                        data[entity_type] = []
+                
+                return {
+                    'manifest': manifest,
+                    'data': data
+                }
+                
+        except Exception as e:
+            logger.error(f"Error parsing package: {str(e)}")
+            raise Exception(f"Failed to parse import package: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)